### DISCUSSION TITLE
How to split large generated file into small files

### DISCUSSION THREAD

**Discussion by aseyfpour** [parcel-bundler/parcel#9005] [Q&A]
I have a problem on using parcel with webextension plugin. The generated background.js file exceeded 4mb size and based on Firefox extension store policy it should be less than 4mb. How can I configure parcel to build ts files with same structure as the source folder or split background.js into smaller files to bypass this issue?

Here is the build output:
```
dist/manifest.json                         828 B     383ms
dist/background.3ee6c422.js          ⚠️  4.68 MB    19.87s
dist/new-tab/index.html                    482 B     527ms
dist/new-tab/index.5d021706.js          13.05 KB     3.58s
dist/new-tab/index.2a8427a2.css          6.67 KB     220ms
dist/popup/index.eb6801dd.js           494.49 KB     2.08s
dist/dashboard/index.7459b276.js        78.19 KB    19.87s
dist/dashboard/index.html                  558 B     526ms
dist/dashboard/index.d5025751.js       672.78 KB     3.58s
dist/popup/index.82fe3cc7.css           31.25 KB     218ms
dist/dashboard/index.b78361a5.css       14.73 KB     386ms
dist/popup/index.html                      519 B    19.90s
dist/popup/index.57bd9afb.js             6.46 KB     493ms
```

I tested with parcel v2.4.1 and also v2.8.3 but both of them have almost same output.

Project repository on github: https://github.com/swashapp/extension

**Comment by joshamaju** [comment]
> How can I configure parcel to build ts files with same structure as the source folder

You're to use a namer plugin for that. Check the plugins page you'll find many
